src/NzbDrone.Core/MediaFiles/TrackImport/Manual/ManualImportService.cs

```diff
@@ -164,109 +166,108 @@ private bool SceneSource(Series series, string folder)
-            //                                        TrackedDownload = trackedDownload,
-            //                                        ImportResult = importResult
-            //                                    });
-            //    }
-            //}
-
-            //_logger.ProgressTrace("Manually imported {0} files", imported.Count);
-
-            //foreach (var groupedTrackedDownload in importedTrackedDownload.GroupBy(i => i.TrackedDownload.DownloadItem.DownloadId).ToList())
-            //{
-            //    var trackedDownload = groupedTrackedDownload.First().TrackedDownload;
-
-            //    if (_diskProvider.FolderExists(trackedDownload.DownloadItem.OutputPath.FullPath))
-            //    {
-            //        if (_downloadedEpisodesImportService.ShouldDeleteFolder(
-            //                new DirectoryInfo(trackedDownload.DownloadItem.OutputPath.FullPath),
-            //                trackedDownload.RemoteEpisode.Series) && !trackedDownload.DownloadItem.IsReadOnly)
-            //        {
-            //            _diskProvider.DeleteFolder(trackedDownload.DownloadItem.OutputPath.FullPath, true);
-            //        }
-            //    }
-
-            //    if (groupedTrackedDownload.Select(c => c.ImportResult).Count(c => c.Result == ImportResultType.Imported) >= Math.Max(1, trackedDownload.RemoteEpisode.Episodes.Count))
-            //    {
-            //        trackedDownload.State = TrackedDownloadStage.Imported;
-            //        _eventAggregator.PublishEvent(new DownloadCompletedEvent(trackedDownload));
-            //    }
-            //}
+
+            var imported = new List<ImportResult>();
+            var importedTrackedDownload = new List<ManuallyImportedFile>();
+
+            for (int i = 0; i < message.Files.Count; i++)
+            {
+                _logger.ProgressTrace("Processing file {0} of {1}", i + 1, message.Files.Count);
+
+                var file = message.Files[i];
+                var artist = _artistService.GetArtist(file.ArtistId);
+                var album = _albumService.GetAlbum(file.AlbumId);
+                var tracks = _trackService.GetTracks(file.TrackIds);
+                var parsedTrackInfo = Parser.Parser.ParseMusicPath(file.Path) ?? new ParsedTrackInfo();
+                var mediaInfo = _videoFileInfoReader.GetMediaInfo(file.Path);
+                var existingFile = artist.Path.IsParentPath(file.Path);
+
+                var localTrack = new LocalTrack
+                {
+                    ExistingFile = false,
+                    Tracks = tracks,
+                    MediaInfo = mediaInfo,
+                    ParsedTrackInfo = parsedTrackInfo,
+                    Path = file.Path,
+                    Quality = file.Quality,
+                    Artist = artist,
+                    Album = album,
+                    Size = 0
+                };
+
+                //TODO: Cleanup non-tracked downloads
```

Comment:
Does this need to be done before merge?

Reply:
Comes from Sonarr, not sure what cleanup there is to do or what this was referring to.